### PR TITLE
Migrate to our Standard CI

### PR DESCRIPTION
**Pull request by DJMcNab**
See also linebender/peniko#21
I haven't added an MSRV yet, as we're not at that stage

This also lets the headless example use the `--use-cpu` flag. This isn't currently used in CI - I'm planning to make a `tests` crate/PR instead

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,55 +1,117 @@
...
-      # - run: cargo run_wasm -- -p with_winit --bin with_winit_bin --build-only
-      - name: Allow using WebGPU in web_sys
-        run: |
-          echo "RUSTFLAGS=--cfg=web_sys_unstable_apis" >> "$GITHUB_ENV"
-      - run: cargo check -p with_winit --target wasm32-unknown-unknown
+          components: clippy
+
+      - name: cargo clippy (wasm)
+        run: cargo clippy --all-targets --target wasm32-unknown-unknown -- -D warnings
+        env:
```
Comment (waywardmonkeys) [inline .github/workflows/ci.yml:97]: This is an improvement!

File: crates/shaders/src/types.rs [hunk 2]
```diff
@@ -20,6 +20,10 @@ pub enum BindType {
 }
 
 impl BindType {
+    // TODO: This is a public method, which means it definitely is not
+    // "dead code". However, rustc seems insitent that it is, and so to not
+    // block forward progress, I shall humour it
+    #[allow(dead_code)]
```
Comment (DJMcNab) [inline crates/shaders/src/types.rs:26]: See https://xi.zulipchat.com/#narrow/stream/255911-rust/topic/Confusing.20method.20is.20never.20used.20warning

I can't prioritise debugging and resolving this, so think we should land this without that

Comment (Philipp-M) [reply crates/shaders/src/types.rs:26]: Yeah I honestly think it's a bug in Rust, the error warning doesn't make sense to me as the type `BindType` is clearly public...

**Review by Philipp-M — APPROVED** [review]
Going towards a more unified CI makes sense. LGTM, thanks :)

**Pull request merged**